tests/sqlparser_common.rs

```diff
@@ -11432,3 +11432,24 @@ fn test_any_some_all_comparison() {
     verified_stmt("SELECT c1 FROM tbl WHERE c1 <> SOME(SELECT c2 FROM tbl)");
     verified_stmt("SELECT 1 = ANY(WITH x AS (SELECT 1) SELECT * FROM x)");
 }
+
+#[test]
+fn test_parse_alter_table_update() {
+    let dialects = all_dialects_where(|d| d.supports_alter_table_update());
```

Comment:
Since the change introduces a new node in the AST, one style of test I think could be useful is to assert the returned AST that it matches what we expect. i.e a test  in this style for [alter_policy](https://github.com/apache/datafusion-sqlparser-rs/blob/17080f1994689009a319cbb8b08314e7c70c228b/tests/sqlparser_common.rs#L11340-L11358)
We can take one scenario like `ALTER TABLE t UPDATE col1 = 1, col2 = 2 IN PARTITION abc WHERE col3 = 1` and verify that the `assignments`, `partition` and `selection` are where we expect them to be in the node